index.bs

```diff
@@ -6583,6 +6737,360 @@ This extension enables use of a user verification method.
+
+    1. Let the `devicePubKey` [=authenticator extension output=] value be a byte string that encodes a [=CBOR=] map as defined by `attObjForDevicePublicKey` above, with keys and values as follows:
+
+        Note: as with all CBOR structures used in this specification, the [=CTAP2 canonical CBOR encoding form=] MUST be used.
+
+        1. Let the `aaguid` key's value be |attAaguid|.
+
+        1. Let the `dpk` key's value be |dpk|.
+
+        1. Let the `nonce` key's value be |randomNonce|.
+
+        1. Let the `scope` key have the value zero (0x00) if this is an "entire device" [=device public key=]. Otherwise, let `scope` have the value one (0x01), indicating a more narrow per-app scope.
+
+        1. Let the values of the `$$attStmtType` [=group socket=] be the result of generating an [=attestation statement=] with the [=attestation statement format=], |attFormat|. See [[#sctn-device-publickey-attestation-calculations]].
+
+            Note: The details of the `$$attStmtType` values are dependent upon the particular [=attestation statement=] format. See [[#sctn-attestation-formats]]. Since the [=hardware-bound device key pair=] is specific to a particular authenticator, its attestation can be tied to hardware roots of trust (although they do not have to be). This is in contrast to the associated [=user credential=]'s attestation, if it is a [=multi-device credential=].
+
+        1. If the `$$attStmtType` [=group socket=] contains uniquely identifying information then let `epAtt` be [TRUE]. Otherwise omit the `epAtt` field. (This field can only be [TRUE] when the {{AuthenticationExtensionsDevicePublicKeyInputs/attestation}} field of the extension input is "enterprise" and either the user-agent or the authenticator permits uniquely identifying attestation for the requested RP ID.)
+
+    1. Let |dpkSig| be the result of signing the [=assertion signature=] [input](#fig-signature) with |devicePrivateKey|.
+
+    1. Output |dpkSig| as the extension's [=unsigned extension output=].
+
+        Note: |dpkSig| cannot be included in the [=authenticator extension output=] because it is returned inside the [=authenticator data=] and that would imply that the signature signs over itself.
+
+##### Attestation calculations ##### {#sctn-device-publickey-attestation-calculations}
+
+When computing attestations, the process in [[#sctn-generating-an-attestation-object]] takes two inputs: `authData` and `hash`. When calculating an attestation for a [=device-bound key=], the typical value for `hash` hashes over the attestation signature itself, which is impossible. Also the attestation of a [=device-bound key=] is potentially used repeatedly, thus may want to be cached. But signing over values that include [=[RP]=]-chosen nonces, like the [=hash of the serialized client data=], make that impossible.
```

Comment:
>the typical value for `hash` hashes over the attestation signature itself

This is no longer true now that `dpkSig` has moved to the unsigned extension outputs, right? (It would be true for the signed extension output)

I think this might instead need to say that the reason is that the `authData` flags and signature counter could change between getAssertions, and therefore the attestation signature can't use `authData` if it's going to be cachable. Because there isn't a circular dependency anymore, right?

Reply:
> This is no longer true now that dpkSig has moved to the unsigned extension outputs, right? (It would be true for the signed extension output)

But the _attestation_ signature is still in the authenticator data. It's the signature by `dpk` that has moved out, but the signature _over_ dpk is still in there.

Reply:
Ah right, of course. Thanks!